lispify/lispify.py

```diff
@@ -0,0 +1,349 @@
+                return True
+
+    def _paren_content_iter(self, pairs):
+        for k, v in pairs:
+            if v is not None:
+                if k in self.keywords:
+                    yield self._kv_pair(k, v)
+                else:
+                    yield self._kv_pair(k, lispify(v))
+
+
+class LispError(LispType):
+
+    """
+    An error with a symbol. The expected value should be an exception
+    with the class name as desired symbol name and the following attributes:
+    :message  a message for internal debugging (used for Python exceptions)
+    :reply    a reply message that is okay to show users
+              (e.g. death-date for a person that is still alive)
+    """
+
+    priority = MAX_PRIORITY
+
+    def should_parse(self):
+        return isinstance(self.val, Exception)
+
+    def val_str(self):
+        symbol = camel_case_to_lisp_name(type(self.val).__name__)
+        kw = self.val.__dict__
+        if self.val.message:
+            kw.update({'message': self.val.message})
+
+        if kw:
+            return u"(:error {symbol} {keys})".format(
+                symbol=symbol,
+                keys=self._plist(kw.items())
+            )
+        else:
+            return u"(:error %s)" % symbol
+
+    def _plist(self, pairs):
+        return " ".join(list(self._paren_content_iter(pairs)))
+
+    def _kv_pair(self, k, v):
+        if k is None:
+            return u"%s" % v
+
+        if isinstance(k, basestring):
+            return u":%s %s" % (k, v)
+
+    def _paren_content_iter(self, pairs):
+        for k, v in pairs:
+            if v is not None:
+                yield self._kv_pair(k, lispify(v))
+
+    def __nonzero__(self):
+        """
+        Errors should count as zeros so that you can check for an answer with
+
+        if potential_error_or_none:
+            # do stuff
+        """
+
+        return False
+
+
+class _LispLiteral(LispType):
+
+    """
+    Lisp literals. These are not.
+    """
+    priority = MAX_PRIORITY
+    literal = True
+
+
+class LispKeyword(_LispLiteral):
+
+    """
+    Just a keyword. No content.
+    """
+
+    def should_parse(self):
+        return isinstance(self.val, basestring) and self.val.startswith(':') \
+            and ' ' not in self.val
+
+    def val_str(self):
+        return self.val
+
+
+class LispBool(_LispLiteral):
+
+    """
+    Lispify a boolean value
+    """
+
+    def should_parse(self):
+        return isinstance(self.val, bool)
+
+    def val_str(self):
+        return u't' if self.val else u'nil'
+
+
+class LispNone(_LispLiteral):
+
+    """
+    Lispified none object
+    """
+
+    def should_parse(self):
+        return self.val is None
+
+    def val_str(self):
+        return u'nil'
+
+
+class LispNumber(_LispLiteral):
+
+    def should_parse(self):
+        return isinstance(self.val, Number)
+
+    def val_str(self):
+        return str(self.val)
+
+
+WIKIBASE_LISP_TYPES = subclasses(LispType, instantiate=False)
```

Comment:
Change the name from `WIKIBASE_LISP_TYPES` to `LISP_TYPES` since lispify is no longer part of Wikipediabase.